### ISSUE TITLE
[BUG] Release builds newer than v0.2 don't run with Ubuntu 20.04

### ISSUE THREAD

**Issue by tadayosi** [redhat-et/microshift#263]
I understand Ubuntu is not currently the main target distro for Microshift, but the pre-built binaries don't run on Ubuntu. I'm trying to run Microshift on GH Actions but it requires Ubuntu distros at this moment. IMO it's beneficial for the Microshift community to support Ubuntu for wider usages.

#### What happened:

Downloading https://github.com/redhat-et/microshift/releases/download/4.7.0-0.microshift-2021-08-31-224727/microshift-linux-amd64 and running it on Ubuntu 20.04 causes the following error:

```
$ ./microshift-linux-amd64 version
microshift-linux-amd64: dl-call-libc-early-init.c:37: _dl_call_libc_early_init: Assertion `sym != NULL' failed.
SIGABRT: abort
PC=0x427644f m=0 sigcode=18446744073709551610
signal arrived during cgo execution

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x4265630, 0xc00009add0, 0x47a36d)
	/usr/lib/golang/src/runtime/cgocall.go:133 +0x5b fp=0xc00009ada0 sp=0xc00009ad68 pc=0x40481b
crypto/internal/boring._Cfunc__goboringcrypto_DLOPEN_OPENSSL(0x0)
	_cgo_gotypes.go:287 +0x4a fp=0xc00009add0 sp=0xc00009ada0 pc=0x6d0cca
crypto/internal/boring.init.0()
	/usr/lib/golang/src/crypto/internal/boring/boring.go:46 +0x4d fp=0xc00009ae08 sp=0xc00009add0 pc=0x6d750d
runtime.doInit(0x7aa8020)
	/usr/lib/golang/src/runtime/proc.go:5652 +0x8a fp=0xc00009ae38 sp=0xc00009ae08 pc=0x448aca
runtime.doInit(0x7aa45a0)
	/usr/lib/golang/src/runtime/proc.go:5647 +0x57 fp=0xc00009ae68 sp=0xc00009ae38 pc=0x448a97
runtime.doInit(0x7ace1c0)
	/usr/lib/golang/src/runtime/proc.go:5647 +0x57 fp=0xc00009ae98 sp=0xc00009ae68 pc=0x448a97
runtime.doInit(0x7acfc80)
	/usr/lib/golang/src/runtime/proc.go:5647 +0x57 fp=0xc00009aec8 sp=0xc00009ae98 pc=0x448a97
runtime.doInit(0x7a91940)
	/usr/lib/golang/src/runtime/proc.go:5647 +0x57 fp=0xc00009aef8 sp=0xc00009aec8 pc=0x448a97
runtime.doInit(0x7acb0a0)
	/usr/lib/golang/src/runtime/proc.go:5647 +0x57 fp=0xc00009af28 sp=0xc00009aef8 pc=0x448a97
runtime.doInit(0x7ac5dc0)
	/usr/lib/golang/src/runtime/proc.go:5647 +0x57 fp=0xc00009af58 sp=0xc00009af28 pc=0x448a97
runtime.doInit(0x7aa1020)
	/usr/lib/golang/src/runtime/proc.go:5647 +0x57 fp=0xc00009af88 sp=0xc00009af58 pc=0x448a97
runtime.main()
	/usr/lib/golang/src/runtime/proc.go:191 +0x1c5 fp=0xc00009afe0 sp=0xc00009af88 pc=0x43b745
runtime.goexit()
	/usr/lib/golang/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc00009afe8 sp=0xc00009afe0 pc=0x471c41

goroutine 19 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x7d53c80)
	/opt/app-root/src/github.com/redhat-et/microshift/vendor/k8s.io/klog/v2/klog.go:1164 +0x8b
created by k8s.io/klog/v2.init.0
	/opt/app-root/src/github.com/redhat-et/microshift/vendor/k8s.io/klog/v2/klog.go:418 +0xdf

rax    0x0
rbx    0x6
rcx    0x427644f
rdx    0x0
rdi    0x2
rsi    0x7fffe5c1a870
rbp    0x59c9510
rsp    0x7fffe5c1a870
r8     0x0
r9     0x7fffe5c1a870
r10    0x8
r11    0x246
r12    0x59ce7ae
r13    0x59ce7c8
r14    0x25
r15    0x91d9f50
rip    0x427644f
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
```

#### What you expected to happen:

`microshift` should run on Ubuntu versions.

#### How to reproduce it (as minimally and precisely as possible):

1. '...'
2. '...'

#### Anything else we need to know?:

Just for info, they don't run on CentOS 7 either (with the same error). Most likely it's because of the glibc versions difference. If I build microshift locally on Ubuntu, it runs fine.

```
$ make GO_REQUIRED_MIN_VERSION:=
...
$ ./microshift version
Microshift Version: 4.7.0-0.microshift-2021-08-31-224727-15-g69538fa
Base OKD Version: 4.7.0-0.okd-2021-06-13-090745
```

#### Environment:
- Microshift version (use `microshift version`):

Any version newer than  v0.2 pre-release.

- Hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):

```
$ uname -a
Linux ubuntu-focal 5.4.0-84-generic #94-Ubuntu SMP Thu Aug 26 20:27:37 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

- Others:

```
$ ldd --version 
ldd (Ubuntu GLIBC 2.31-0ubuntu9.2) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

**Comment by rootfs** [comment]
this is the same issue as #219 cc @jbpratt

**Comment by tadayosi** [comment]
@rootfs @copejon Sorry but may I ask reopening the issue?  The merged pull req #271 is not the fix for this issue. I mentioned this issue from there just because this blocks that pull req from fully functioning otherwise.

Also I'd doubt that this is a duplicate of #219. I think the defining error message of this issue is this line:
> microshift-linux-amd64: dl-call-libc-early-init.c:37: _dl_call_libc_early_init: Assertion `sym != NULL' failed.
SIGABRT: abort
and I don't see the same error in #219 description. Googling the error suggests it's related to glibc static linking.

**Comment by rootfs** [comment]
@tadayosi this message is related to #219 
```
__goboringcrypto_DLOPEN_OPENSSL
```
